src/compat/eme/custom_media_keys/old_webkit_media_keys.ts

```diff
@@ -84,7 +82,7 @@ class OldWebkitMediaKeySession
     this.expiration = NaN;
 
     const onSessionRelatedEvent = (evt: Event) => {
-      this.trigger(evt.type, evt);
+      this.trigger(evt.type as keyof MediaKeySessionEventMap, evt);
```

Comment:
I'm wondering if we can type onSessionRelatedEvent with  `MediaKeyMessageEvent | Event`

Reply:
line 84? What would that change?

Reply:
I don't know how to fix this, if you have an idea, you can commit it

Reply:
I'm not sure about how is it supposed to work on the "old webkit", from what I read, we can have an event
triggered from `error` with a type that is different, it seems weird ?
```js
mediaElement.addEventListener("error", (event) => event.type /* keystatuseschange*/)
```

Reply:
? How is it possible? Where did you see that possibility?

Reply:
I thought that event "type" was more precised in typescript but it's always a string
In fact what we can do is: 
```
    const onSessionRelatedEvent = (evt: MediaKeyMessageEvent | Event) => {
      this.trigger(evt.type, evt);
    };
```

but I'm not sure if it's better than casting

Reply:
Yes I'm kind of lost on that one.

In any case, the `compat` EME files are the "ugliest" TypeScript files in the RxPlayer to me but it makes sense that it is (due to the beauty of EME implementations out there).